ldb/validators.py

```diff
@@ -13,3 +13,33 @@ def validate_ldap_username(username):
         raise ValidationError(
             "User %(username)s does not exist", params={"username": username}
         )
+
+
+def validate_google_username(username: str, person):
+    from ldb.models import Person
+
+    if not username:
+        return
+
+    # Check if the username is not an email address
+    if "@" in username:
+        raise ValidationError(
+            "Username %(username)s is not a valid Google username, please use the part before the @",
+            params={"username": username},
+        )
+
+    # Check if another person has the same username
+    if Person.objects.filter(google_username=username).exclude(pk=person.pk).exists():
+        raise ValidationError(
+            "Google Username %(username)s is already taken",
+            params={"username": username},
+        )
+
+    # Check if the person has an ldap_username and if it is the same as the google_username
+    if person.ldap_username and person.ldap_username != username:
+        raise ValidationError(
+            "Google Username %(username)s does not match the LDAP username %(ldap_username)s",
+            params={"username": username, "ldap_username": person.ldap_username},
+        )
```

Comment:
In de main message stond dat er ook een check gedaan zou worden of het niet de ldap username van een andere gebruiker zou zijn. Die check zit hier nu niet in.